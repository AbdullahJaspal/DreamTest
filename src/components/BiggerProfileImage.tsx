import {
  StyleSheet,
  Image,
  Pressable,
  ImageStyle,
  ImageProps,
} from 'react-native';
import React from 'react';
import {ViewStyle} from 'react-native';
import {icons} from '../assets/icons';

interface BiggerProfileImageProps {
  uri: string;
  width?: number;
  height?: number;
  onPress?: () => void;
  containerStyle?: ViewStyle | undefined;
  pictureStyle?: ImageStyle | undefined;
  coverStyle?: ImageStyle | undefined;
  coverPic?: ImageProps;
}

const BiggerProfileImage: React.FC<BiggerProfileImageProps> = ({
  uri,
  width = 60,
  height = 60,
  onPress,
  containerStyle,
  pictureStyle,
  coverStyle,
  coverPic,
}) => {
  const styles = StyleSheet.create({
    container: {
      ...containerStyle,
    },
    profile_image: {
      width: width,
      height: height,
      borderRadius: 35,
      borderColor: '#020202',
      ...pictureStyle,
    },
    cover_images: {
      width: 110,
      height: 110,
      borderColor: '#020202',
      position: 'absolute',
      top: -26,
      left: -26,
      ...coverStyle,
    },
  });

  return (
    <Pressable style={styles.container} onPress={onPress}>
      <Image
        source={uri ? {uri: uri} : icons.userFilled}
        style={styles.profile_image}
      />

      {coverPic && <Image source={coverPic} style={styles.cover_images} />}
    </Pressable>
  );
};

export default React.memo(BiggerProfileImage);
