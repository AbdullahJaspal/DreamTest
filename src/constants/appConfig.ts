export const supportedLanguages = ['en', 'ar', 'fr', 'de'];
