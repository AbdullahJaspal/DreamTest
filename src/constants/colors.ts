export const customGiftColors: string[] = [
  'rgba(255, 0, 92, 0.3)',
  'rgba(255, 0, 92, 0.25)',
  'rgba(255, 0, 92, 0.2)',
  'rgba(255, 0, 92, 0.15)',
  'rgba(255, 0, 92, 0.1)',
  'rgba(255, 0, 92, 0.05)',
  'rgba(255, 0, 92, 0.0)',
];

export const gradientColors: string[] = [
  'rgba(255, 0, 92, 1)',
  'rgba(255, 0, 92, 0.8)',
  'rgba(255, 0, 92, 0.60)',
  'rgba(255, 0, 92, 0.40)',
  'rgba(255, 0, 92, 0.20)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.0)',
  'rgba(255, 0, 92, 0.2)',
  'rgba(255, 0, 92, 0.4)',
  'rgba(255, 0, 92, 0.6)',
  'rgba(255, 0, 92, 0.8)',
  'rgba(255, 0, 92, 1)',
];

export const hashtagGradient: string[] = [
  '#E6F0FF',
  '#ECF4FF',
  '#F1F7FF',
  '#F7FBFF',
  '#FCFEFF',
  '#FEFFFF',
  '#FEFFFF',
  '#FEFFFF',
  '#FEFFFF',
  '#FFFFFF',
];

export const wheelLuckGradient: string[] = [
  '#fcacac',
  '#fcbbbb',
  '#fcc7c7',
  '#ffd9d9',
  '#fce3e3',
  '#fcf2f2',
  '#faf7f7',
];
