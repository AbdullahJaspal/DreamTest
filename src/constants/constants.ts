// import {Dimensions, Platform, StatusBar} from 'react-native';

// const Visibility = Object.freeze({
//   Hidden: 0,
//   Visible: 1,
// });

// const TESTING_PIC_URL = 'https://avatars.githubusercontent.com/u/129264144';

// const SERVER_DOMAIN = 'https://api.dreamlived.com';
// const SERVER_API_URL = 'https://api.dreamlived.com';

// const LIVE_STREAM_URL = 'rtmp://192.168.29.68';

// const AGORA_APP_ID = '317633f93b81adc8a92fc623c4917f5';

// const KEY_STORAGE = Object.freeze({
//   TOKEN: 'token',
//   ID_USER: 'id_user',
//   SEARCH_HIS: 'search_his',
// });

// const {width: WIDTH, height: HEIGHT} = Dimensions.get('screen');
// const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : StatusBar.currentHeight;

// const BOTTOM_NAVIGATOR_HEIGHT = HEIGHT - Dimensions.get('window').height;

// export {
//   Visibility,
//   SERVER_API_URL,
//   SERVER_DOMAIN,
//   KEY_STORAGE,
//   WIDTH,
//   HEIGHT,
//   STATUSBAR_HEIGHT,
//   BOTTOM_NAVIGATOR_HEIGHT,
//   LIVE_STREAM_URL,
//   AGORA_APP_ID,
//   TESTING_PIC_URL,
// };

import {Dimensions, Platform, StatusBar} from 'react-native';

const Visibility = Object.freeze({
  Hidden: 0,
  Visible: 1,
});

const TESTING_PIC_URL = 'https://avatars.githubusercontent.com/u/129264144';

const SERVER_DOMAIN = 'https://api.dreamlived.com';
const SERVER_API_URL = 'https://api.dreamlived.com';

const LIVE_STREAM_URL = 'rtmp://192.168.29.68';

const AGORA_APP_ID = '317633f93b81adc8a92fc623c4917f5';

const KEY_STORAGE = Object.freeze({
  TOKEN: 'token',
  ID_USER: 'id_user',
  SEARCH_HIS: 'search_his',
});

const {width: WIDTH, height: HEIGHT} = Dimensions.get('screen');
const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : StatusBar.currentHeight;

const BOTTOM_NAVIGATOR_HEIGHT = HEIGHT - Dimensions.get('window').height;
export {
  Visibility,
  SERVER_API_URL,
  SERVER_DOMAIN,
  KEY_STORAGE,
  WIDTH,
  HEIGHT,
  STATUSBAR_HEIGHT,
  BOTTOM_NAVIGATOR_HEIGHT,
  LIVE_STREAM_URL,
  AGORA_APP_ID,
  TESTING_PIC_URL,
};
