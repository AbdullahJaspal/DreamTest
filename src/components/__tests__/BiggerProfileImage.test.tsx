import 'react-native';
import React from 'react';

import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import BiggerProfileImage from '../BiggerProfileImage';
import {TESTING_PIC_URL} from '../../constants/constants';

it('Container renders correctly', () => {
  // Render the CText component
  const tree = renderer
    .create(
      <BiggerProfileImage
        uri={TESTING_PIC_URL}
        width={0}
        height={0}
        onPress={function (): void {
          console.log('Bigger Profile click');
        }}
        containerStyle={undefined}
        pictureStyle={undefined}
        coverStyle={undefined}></BiggerProfileImage>,
    )
    .toJSON();

  // Compare the rendered output to the saved snapshot
  expect(tree).toMatchSnapshot();
});
