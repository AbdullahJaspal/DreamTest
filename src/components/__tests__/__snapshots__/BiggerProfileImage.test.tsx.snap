// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={{}}
>
  <Image
    source={
      {
        "uri": "https://avatars.githubusercontent.com/u/129264144",
      }
    }
    style={
      {
        "borderColor": "#020202",
        "borderRadius": 35,
        "height": 0,
        "width": 0,
      }
    }
  />
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/crown.png",
      }
    }
    style={
      {
        "borderColor": "#020202",
        "height": 90,
        "left": -15,
        "position": "absolute",
        "top": -14,
        "width": 90,
      }
    }
  />
</View>
`;
